### PR TITLE
Support of the `shape` argument in `empty_like()` & Co.

### PR DESCRIPTION
**Pull request by madsbk**
Implements the first part of https://github.com/nv-legate/cunumeric/issues/840.

### CODE REVIEW

File: tests/integration/test_array_creation.py [hunk 1]
```diff
@@ -137,21 +137,26 @@ def test_full_bad_filled_value(self):
...
     assert xfl.shape == yfl.shape
     assert xfl.dtype == yfl.dtype
 
 
 @pytest.mark.parametrize("x_np,dtype", DATA_ARGS)
 @pytest.mark.parametrize("fn", LIKE_FUNCTIONS)
-def test_func_like(fn, x_np, dtype):
+@pytest.mark.parametrize("shape", SHAPE_ARG)
+def test_func_like(fn, x_np, dtype, shape):
+    shape = shape if shape is None else x_np.reshape(shape).shape
```
Comment (manopapad) [inline tests/integration/test_array_creation.py:159]: The `shape` is not actually used in this test (and the one below?)

Comment (madsbk) [reply tests/integration/test_array_creation.py:159]: Oops, forgot to commit, fixed.

File: tests/integration/test_array_creation.py [hunk 2]
```diff
@@ -137,39 +137,46 @@ def test_full_bad_filled_value(self):
...
 ]
 LIKE_FUNCTIONS = ("zeros_like", "ones_like")
+SHAPE_ARG = (None, (-1,), (1, -1))
 
 
 @pytest.mark.parametrize("x_np,dtype", DATA_ARGS)
-def test_empty_like(x_np, dtype):
+@pytest.mark.parametrize("shape", SHAPE_ARG)
+def test_empty_like(x_np, dtype, shape):
+    shape = shape if shape is None else x_np.reshape(shape).shape
```
Comment (manopapad) [inline tests/integration/test_array_creation.py:146]: I don't think you need to do this. `x_np.reshape(shape)` doesn't modify `x_np`, it creates a new view/copy. And `arr.reshape(s).shape` always returns `s`.

Comment (madsbk) [reply tests/integration/test_array_creation.py:146]: It is just a convenience way to create compatible shapes. 
Notice, `arr.reshape(s).shape` makes NumPy/cuNumeric infer the value of the `-1` in `s`.

Comment (manopapad) [reply tests/integration/test_array_creation.py:146]: Ah, ok, I missed the -1 in the shapes. You're right, those need to be sanitized.

**Review by manopapad — APPROVED** [review]

**Comment by madsbk** [comment]
Anything else I need to do here?

**Pull request merged**